src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/RandomData.kt

```diff
@@ -33,17 +36,50 @@ fun randomInt(min: Int, max: Int) = Random.nextInt(min, max)
 fun randomLong() = Random.nextLong(Long.MIN_VALUE, Long.MAX_VALUE)
 fun randomDouble(min: Double, max: Double) = Random.nextDouble(min, max)
 
+fun LocalDate.nonRepeatingRandomDateAfter(groupingKey: String, maxDays: Int): LocalDate = NonRepeatingRandomValueManager.getInstance().generateUniqueValue(groupingKey) {
+  this.randomDateAfter(maxDays)
+}
+
 fun LocalDate.randomDateAfter(maxDays: Int): LocalDate = this.plusDays(randomInt(1, maxDays).toLong())
 fun LocalDate.randomDateBefore(maxDays: Int): LocalDate = this.minusDays(randomInt(1, maxDays).toLong())
 fun LocalDate.randomDateAround(maxDays: Int): LocalDate = this.minusDays(maxDays.toLong()).randomDateAfter(maxDays * 2)
 
-fun LocalDateTime.randomDateTimeAfter(maxDays: Int): LocalDateTime = this.plusDays(randomInt(1, maxDays).toLong())
 fun LocalDateTime.randomDateTimeBefore(maxDays: Int): LocalDateTime = this.minusDays(randomInt(1, maxDays).toLong())
 
 fun OffsetDateTime.randomDateTimeAfter(maxDays: Int): OffsetDateTime = this.plusMinutes(randomInt(1, 60 * 24 * maxDays).toLong()).truncatedTo(ChronoUnit.SECONDS)
 fun OffsetDateTime.randomDateTimeBefore(maxDays: Int): OffsetDateTime = this.minusMinutes(randomInt(1, 60 * 24 * maxDays).toLong()).truncatedTo(ChronoUnit.SECONDS)
 
-fun Instant.randomDateTimeAfter(maxDays: Int): Instant = this.plus(randomInt(1, 60 * 24 * maxDays).toLong(), ChronoUnit.MINUTES).truncatedTo(ChronoUnit.SECONDS)
 fun Instant.randomDateTimeBefore(maxDays: Int): Instant = this.minus(randomInt(1, 60 * 24 * maxDays).toLong(), ChronoUnit.MINUTES).truncatedTo(ChronoUnit.SECONDS)
 
 fun <T> randomOf(options: List<T>) = options[randomInt(0, options.size - 1)]
+
+@Service
+class NonRepeatingRandomValueManager(applicationContext: ApplicationContext) {
+  private val generatedValuesByKey = mutableMapOf<String, List<*>>()
```

Comment:
Minor observation: since we are using unique values, we could consider using Set rather than List here?

Reply:
Good idea, done 👍